test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,591 @@
+
+		})
+		g.It("and ensure default gatewayclass is accepted", func() {
+
+			g.By("Check if default GatewayClass is accepted after OLM resources are successful")
+			errCheck := checkGatewayClass(oc, gatewayClassName)
+			o.Expect(errCheck).NotTo(o.HaveOccurred())
+			e2e.Logf("GatewayClass %s successfully installed and accepted!", gatewayClassName)
+
+		})
+		g.It("and ensure custom gatewayclass can be accepted", func() {
+			gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+
+			g.By("Create Custom GatewayClass")
+			gatewayClass := buildGatewayClass("custom-gatewayclass", gatewayClassControllerName)
+			gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+			if err != nil {
+				e2e.Logf("Gateway Class %s already exists, or has failed to be created, checking its status", "custom-gatewayclass")
+			}
+			errCheck := checkGatewayClass(oc, "custom-gatewayclass")
+			o.Expect(errCheck).NotTo(o.HaveOccurred())
+			e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+			g.By("Deleting Custom GatewayClass and confirming that it is no longer there")
+			err = gwapiClient.GatewayV1().GatewayClasses().Delete(context.Background(), "custom-gatewayclass", metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			_, err = gwapiClient.GatewayV1().GatewayClasses().Get(context.Background(), "custom-gatewayclass", metav1.GetOptions{})
+			o.Expect(err).To(o.HaveOccurred())
+			e2e.Logf("The custom gatewayClass %s has been sucessfully deleted", "custom-gatewayclass")
+
+			g.By("check if default gatewayClass is accepted and ISTIO CR and pod are still available")
+			defaultCheck := checkGatewayClass(oc, gatewayClassName)
+			o.Expect(defaultCheck).NotTo(o.HaveOccurred())
+
+			waitIstio := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+				deployIstio, err := oc.AsAdmin().CoreClient().AppsV1().Deployments("openshift-ingress").Get(context, "istiod-openshift-gateway", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if deployIstio.Status.ReadyReplicas != 1 {
+					e2e.Logf("Deployment Pod %s is not ready, retrying...", "istiod-openshift-gateway")
+					return false, nil
+				}
+				e2e.Logf("Deployment Pod %s ready", "istiod-openshift-gateway")
+				return true, nil
+			})
+			if waitIstio != nil {
+				e2e.Failf("Istio Deployment Pod never got ready")
+			}
+
+			g.By("Confirm that ISTIO CR is created and in healthy state")
+			resource := types.NamespacedName{Namespace: "openshift-ingress", Name: "openshift-gateway"}
+
+			istioStatus, errIstio := oc.AsAdmin().Run("get").Args("-n", resource.Namespace, "istio", resource.Name, "-o=jsonpath={.status.state}").Output()
+			o.Expect(errIstio).NotTo(o.HaveOccurred())
+			o.Expect(istioStatus).To(o.Equal(`Healthy`))
+		})
+
+		g.It("and negative tests for OSSM related resources", func() {
```

Comment:
The word `negative tests` is unclear, could you rename to `and ensure OSSM subscription and istio gets recreated after deleting them`

Reply:
done, thanks!